### DISCUSSION TITLE
Dynamically Updated Time Strip Data from Back-End

### DISCUSSION THREAD

**Discussion by Cameronseitz** [nasa/openmct#4317] [Community Q&A]
I was wondering if anyone knew how you can upload JSON from the back-end from a Rest API or socket connection, NOT manually from the GUI, and how you would update the time strip based on live telemetry from a back-end source. Also, if updating based on live telemetry is not possible, can you at least update the time strip color properties from the back-end source? This seems like something Open MCT has functionality for, but not sure how to go about doing it.  Lastly, for demo purposes, how do you indicate the start of an event i.e. start all the events starting with the first few? If anyone knows the answer to any of these, help would be much appreciated. Thanks!

**Comment by jvigliotta** [comment]
@shefalijoshi could you help @Cameronseitz with this?

**Comment by pharapeti** [comment]
You should be able to achieve such a solution by using WebSockets (check out socket.io or [vue-socket.io](https://github.com/MetinSeylan/Vue-Socket.io)) to subscribe to a socket server hosted by your backend. Then once you upload the JSON file to your backend server, you would send a socket message from the backend to your OpenMCT frontend with the relevant data to update your time bounds - where you would interact with the OpenMCT Time API to update the bounds.

For example:
```
socket.on('updateTimeBound', (data) => {
  openmct.time.bounds({ start: startTime end: endTime);
});
```

**Comment by akhenry** [comment]
@shefalijoshi

**Comment by shefalijoshi** [comment]
@Cameronseitz I need some clarification before I can suggest a solution here. By timestrip are you referring to a Gantt chart or a Plan?
Are you using any backend databases in conjunction with Open MCT?
Answers to these questions would greatly help guide you in the right direction.
Thanks!

**Comment by akhenry** [comment]
Have created an issue to define some documentation for Activity timelines - https://github.com/nasa/openmct/issues/8166
